packages/office-ui-fabric-react/src/utilities/grid/Grid.tsx

```diff
@@ -0,0 +1,136 @@
+import * as React from 'react';
+import { BaseComponent } from '../../Utilities';
+import { IGridProps } from './Grid.Props';
+
+export interface IGridState {
+
+}
+
+export class Grid extends BaseComponent<IGridProps, IGridState> {
+
+  // Used to track the index of item we are on while building grid
+  private _gridIndex;
+
+  private _id;
+
+  constructor(props: IGridProps) {
+    super(props);
+    this._gridIndex = 0;
+  }
+
+  public render() {
+    let {
+      items,
+      width,
+      onRenderItem,
+      positionInSet,
+      setSize
+    } = this.props;
+
+    // Reset the index so we will start at the beginning
+    this._gridIndex = 0;
+
+    // Create the table/grid
+    return (
+      <table
+        role={ 'grid' }
+        aria-posinset={ positionInSet }
+        aria-setsize={ setSize }
+        style={ { padding: '2px', outline: 'none' } }>
+        <tbody>
+          { this._buildGridContents() }

```

Comment:
it becomes declarative and functional, easy to read and maintain. I don't think it would affect perf, there other things way more expensive than that.